nav2_bringup/launch/nav2_bringup_launch.py

```diff
@@ -14,118 +14,85 @@
+        default_value=os.path.join(
+            get_package_prefix('nav2_bt_navigator'),
             'behavior_trees', 'navigate_w_replanning_and_recovery.xml'),
         description='Full path to the behavior tree xml file to use')
 
-    start_map_server_cmd = launch_ros.actions.Node(
-        package='nav2_map_server',
-        node_executable='map_server',
-        node_name='map_server',
-        output='screen',
-        parameters=[configured_params])
-
-    start_localizer_cmd = launch_ros.actions.Node(
-        package='nav2_amcl',
-        node_executable='amcl',
-        node_name='amcl',
-        output='screen',
-        parameters=[configured_params])
-
-    start_dwb_cmd = launch_ros.actions.Node(
-        package='dwb_controller',
-        node_executable='dwb_controller',
-        output='screen',
-        parameters=[configured_params])
-
-    start_planner_cmd = launch_ros.actions.Node(
-        package='nav2_planner',
-        node_executable='planner_server',
-        node_name='planner_server',
-        output='screen',
-        parameters=[configured_params])
-
-    start_recovery_cmd = launch_ros.actions.Node(
-        package='nav2_recoveries',
-        node_executable='recoveries_node',
-        node_name='recoveries',
-        output='screen',
-        parameters=[{'use_sim_time': use_sim_time}])
-
-    start_navigator_cmd = launch_ros.actions.Node(
-        package='nav2_bt_navigator',
-        node_executable='bt_navigator',
-        node_name='bt_navigator',
-        output='screen',
-        parameters=[configured_params])
+    declare_autostart_cmd = DeclareLaunchArgument(
+        'autostart', default_value='true',
+        description='Automatically startup the nav2 stack')
 
-    start_lifecycle_manager_cmd = launch_ros.actions.Node(
+    # Specify the actions
+    start_localization_cmd = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(os.path.join(launch_dir, 'nav2_localization_launch.py')),
+        launch_arguments={'map_yaml_file': map_yaml_file,
+                          'use_sim_time': use_sim_time,
+                          'autostart': autostart,
+                          'params_file': params_file,
+                          'use_lifecycle_mgr': 'false'}.items())
+
+    start_navigation_cmd = IncludeLaunchDescription(
+        PythonLaunchDescriptionSource(os.path.join(launch_dir, 'nav2_navigation_launch.py')),
+        launch_arguments={'use_sim_time': use_sim_time,
+                          'autostart': autostart,
+                          'params_file': params_file,
+                          'bt_xml_file': bt_xml_file,
+                          'use_lifecycle_mgr': 'false'}.items())
+
+    start_lifecycle_manager_cmd = Node(
         package='nav2_lifecycle_manager',
         node_executable='lifecycle_manager',
         node_name='lifecycle_manager',
         output='screen',
-        parameters=[configured_params])
+        parameters=[{'use_sim_time': use_sim_time},
```

Comment:
Since we're no longer using the params file for lifecycle_manager, can we remove those params from the file?

Reply:
good point, removed the lifecycle mgr params from the file